ddclient.in

```diff
@@ -5787,6 +5800,83 @@ sub nic_gandi_update {
     }
 }
 
+######################################################################
+## nic_keysystems_examples
+######################################################################
+sub nic_keysystems_examples {
+   return <<EoEXAMPLE;
+o 'keysystems'
+
+The 'keysystems' protocol is used by some non-free dynamic DNS services
+such as www.domaindiscount24.com, www.moniker.com or www.rrpproxy.net.
+Check https://www.domaindiscount24.com/faq/en/dynamic-dns for API.
+
+Configuration variables applicable to the 'keysystems' protocol are:
+ protocol=keysystems               ##
+ server=www.fqdn.of.service   ## defaults to dynamicdns.key-systems.net
+ password=service-password    ## password (token) registered with the service
+ non-fully.qualified.host         ## the host registered with the service.
+
+Example ${program}.conf file entries:
+ ## single host update
+ protocol=keysystems,                                       \\
+ password=prettypassword                    \\
+ myhost
+
+EoEXAMPLE
+}
+
+######################################################################
+## nic_keysystems_update
+## By JMCC
+## Based on the work by Tom Bosmans - https://bit.ly/2SZI2zD
+## response contains "code 200" on succesfull completion
+######################################################################
+sub nic_keysystems_update {
+   debug("\nnic_keysystems_update -------------------");
+
+   ## update each configured host
+   ## should improve to update in one pass
+   foreach my $h (@_) {
+       my $ip = delete $config{$h}{'wantip'};
+       info("KEYSYSTEMS setting IP address to %s for %s", $ip, $h);
+       verbose("UPDATE:","updating %s", $h);
+
+       # Set the URL that we're going to to update
+       my $url;
+       $url  = "http://$config{$h}{'server'}/update.php";
+       $url .= "?hostname=";
+       $url .= $h;
+       $url .= "&password=";
+       $url .= $config{$h}{'password'};
+       $url .= "&ip=";
+       $url .= $ip;
```

Comment:
```suggestion
       $url  = "http://$config{$h}{'server'}/update.php";
       $url .= "?hostname=$h";
       $url .= "&password=$config{$h}{'password'}";
       $url .= "&ip=$ip";
```

Reply:
Well, I tested it as it was, and it works. I cannot guarantee it will work with these changes. Are they really necessary?

Reply:
FWIW, I have studied the above change, and it looks good to me; it simply reduces the number of lines of code by merging logically-related bits of string together.